msal/src/main/java/com/microsoft/identity/client/TokenParameters.java

```diff
@@ -40,14 +41,33 @@
     private String mAuthority;
     private ClaimsRequest mClaimsRequest;
     private AccountRecord mAccountRecord;
+    private AuthenticationScheme mAuthenticationScheme;
 
-    protected TokenParameters(final TokenParameters.Builder builder) {
+    protected TokenParameters(@NonNull final TokenParameters.Builder builder) {
         mAccount = builder.mAccount;
         mAuthority = builder.mAuthority;
         mClaimsRequest = builder.mClaimsRequest;
         mScopes = builder.mScopes;
+
+        // In the future, if more scheme are added, some conditional logic will need to be
+        // introduced here.
+        if (null != builder.mAuthenticationScheme) {
```

Comment:
I think this logic can move to the `OperationParametersAdapter` class. I think TokenParameters should just contain the value that was set by the user (which could be `null` if the user didn't provide anything), and then when creating `OperationParameters` we can set the appropriate value (or the fallback value) based on this logic.

Reply:
I'll take this feedback, I am going to update the `AuthenticationSchemeFactory` in `common` (which is called by `OperationParametersAdapter`)

Reply:
Resolved in 757354f0202ec40b61377c76386ad0948fd0ded1